library/lib_pid.ks

```diff
@@ -53,22 +57,23 @@ function PID_seek {
     // I have never been called yet - so don't trust any
     // of the settings yet.
   } else {
-    if dT = 0 { // Do nothing if no physics tick has passed from prev call to now.
-      set newInput to oldInput.
-    } else {
-      set D to (P - oldP)/dT. // crude fake derivative of P
+    if dT > 0 { // Do nothing if no physics tick has passed from prev call to now.
+     set D to (P - oldP)/dT. // crude fake derivative of P
+     local onlyPD is Kp*P + Kd*D.
+     if (oldI > 0 or onlyPD > cMin) and (oldI < 0 or onlyPD < cMax) { // only do the I turm when within the control range
       set I to oldI + P*dT. // crude fake integral of P
-      set newInput to Kp*P + Ki*I + Kd*D.
+     }.
+     set newInput to onlyPD + Ki*I.
     }.
   }.
 
```

Comment:
Can't we insert this here:
```
  set newInput to max(cMin,min(cMax,newInput)).
```
It's not required to fix the roundup, but it doesn't hurt and it helps make things simpler to explain.  You don't need to explain the fact that cMin and cMax don't really limit the range returned.  Also it lets people tell the PID controller, if they like, that they only allow it to use a limited control authority (i.e. like telling it it can only do RCS translation of no more than +/- 0.25 or so, if they want).